vhost-device-spi/README.md

```diff
@@ -0,0 +1,52 @@
+# vhost-device-spi - SPI emulation backend daemon
+
+## Description
+This program is a vhost-user backend that emulates a VirtIO SPI bus.
+This program takes the layout of the spi bus and its devices on the host
+OS and then talks to them via the /dev/spidevX.Y interface when a request
+comes from the guest OS for a SPI device.
+
+## Synopsis
+
+**vhost-device-spi** [*OPTIONS*]
+
+## Options
+
+.. program:: vhost-device-spi
+
+.. option:: -h, --help
+
+  Print help.
+
+.. option:: -s, --socket-path=PATH
+
+  Location of vhost-user Unix domain sockets, this path will be suffixed with
+  0,1,2..socket_count-1.
+
+.. option:: -c, --socket-count=INT
+
+  Number of guests (sockets) to attach to, default set to 1.
+
+.. option:: -l, --device=SPI-DEVICES
+
+  Spi device full path at the host OS in the format:
+      /dev/spidevX.Y
+
+  Here,
+      X: is spi controller's bus number.
+      Y: is chip select index.
+
+## Examples
+
+The daemon should be started first:
```

Comment:
Should we also put an example on how to start a VMM (e.g. QEMU)?

I would like to give it a try but I have no idea how to use it.

Reply:
Actually I verify this project on my target, with the VMM which is not open source. Now I prepare the QEMU environment, and it will take some time.

Reply:
I have verified the following commend works:
```
qemu-system
-chardev socket,path=vspi.sock0,id=vspi
-device vhost-user-spi-pci,chardev=vspi,id=spi
-m 4096
...
```

Now I need to submit a qemu patch for supporting vhost-user-spi, and it will also take some time.